### DISCUSSION TITLE
problme about Martensite Parent Grain Reconstruction

### DISCUSSION THREAD

**Discussion by lwlgeo** [mtex-toolbox/mtex#1526] [Ask Anything]
I just find the new function to reconstruct the Austinite phase by using the Martensite Parent Grain Reconstruction fuction in MTEX 5.8.1. it works well on the given data, but not on my ebsd data. about 50% Martensite grains are not reconstructed correctly? what is the problem?  any help?

the code is from the MTEX 5.8.1,  nothing change. My data is also uploaded. the final reconstructed map is also uploaded. Thanks a lot!

![untitled](https://user-images.githubusercontent.com/32723243/184276079-44470beb-e63f-44aa-9588-1d25a81c1df3.png)
Data:https://1drv.ms/u/s!AtSX8jEOLmVajjbNxpK8SliZQfr2?e=rm3R5V

**Comment by nyyssont** [comment]
Hello,

some points:
1. There appears to be some type of ferrite in the orientation map. This should not be reconstructed but separated from the rest of the data.
2. There is some drift in a portion of the map. This sort of thing can affect reconstruction as well.

However, the code below reconstructed your data nicely. Note that I have not separated the ferrite from the analysis.

```
% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('432', [2.9 2.9 2.9], 'mineral', 'Ferrite', 'color', [0.53 0.81 0.98]),...
  crystalSymmetry('432', [3.6 3.6 3.6], 'mineral', 'Austenite', 'color', [0.56 0.74 0.56])};

% plotting convention
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% which files to be imported
fname = '1040.ang';

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ang',...
  'convertEuler2SpatialReferenceFrame');

%%
[grains, ebsd.grainId] = calcGrains(ebsd)

ebsd(grains(grains.grainSize < 2)).rotations = quaternion.nan;
ebsd(grains(grains.grainSize < 2)).phase = 0;
ebsd(grains(grains.grainSize < 2)).grainId = 0;

ebsd.grainId = zeros(size(ebsd));

[grains,ebsd('indexed').grainId] = calcGrains(ebsd('indexed'),'angle',3*degree);

%%
p2c = orientation.KurdjumovSachs(ebsd('Austenite').CS,ebsd('Ferrite').CS);
job = parentGrainReconstructor(ebsd,grains,p2c) % define job
job.useBoundaryOrientations = true;
% clearvars grains ebsd

%%
job.calcParent2Child('quantile',0.6,'c2c'); % get representative OR
%%
job.calcVariantGraph('threshold',5*degree)
job.clusterVariantGraph('numIter',3,'inflationPower', 1)
job.calcParentFromVote

%%
ebsd_aus = job.calcParentEBSD;
ebsd_aus.grainId = zeros(length(ebsd_aus),1);
%%
[grains_aus,ebsd_aus('Austenite').grainId] = calcGrains(ebsd_aus('Austenite'),'angle',10*degree);

```

![image](https://user-images.githubusercontent.com/10060198/184672216-b791aac6-7536-4981-b88a-8733c2fc21bd.png)

**Reply by lwlgeo** [reply]
Hi, nyyssont

The code run very well, and helps a lot. thank you very much. 

Best,
